src/profile/v100/app_preferences/resource_preferences/resource_preferences.rs

```diff
@@ -0,0 +1,268 @@
+#[derive(
+    Debug,
+    Default,
+    Serialize,
+    Deserialize,
+    PartialEq,
+    Eq,
+    Clone,
+    derive_more::Display,
+    uniffi::Record,
+)]
+#[serde(rename_all = "camelCase")]
+#[display("{}", self.description())]
+pub struct ResourcePreferences {
+    /// A dictionary detailing the user preferences for fungible resources.
+    #[serde(default)]
+    pub fungible: HashMap<ResourceAddress, EntityFlags>,
+
+    /// A dictionary detailing the user preferences for non-fungible resources.
+    #[serde(default)]
+    pub non_fungible: HashMap<NonFungibleGlobalId, EntityFlags>,
+
+    /// A dictionary detailing the user preferences for pool units.
+    #[serde(default)]
+    pub pool_unit: HashMap<PoolAddress, EntityFlags>,
+}
+
+impl Hash for ResourcePreferences {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        let mut fungible_pairs: Vec<_> = self.fungible.iter().collect();
+        fungible_pairs.sort_by_key(|i| i.0);
+
+        let mut non_fungible_pairs: Vec<_> = self.non_fungible.iter().collect();
+        non_fungible_pairs.sort_by_key(|i| i.0);
+
+        let mut pool_unit_pairs: Vec<_> = self.pool_unit.iter().collect();
+        pool_unit_pairs.sort_by_key(|i| i.0);
+
+        Hash::hash(&fungible_pairs, state);
+        Hash::hash(&non_fungible_pairs, state);
+        Hash::hash(&pool_unit_pairs, state);
+    }
+}
+
+impl ResourcePreferences {
+    pub fn new() -> Self {
+        Self {
+            fungible: HashMap::new(),
+            non_fungible: HashMap::new(),
+            pool_unit: HashMap::new(),
+        }
+    }
+
+    pub fn description(&self) -> String {
+        format!(
+            r#"
+            fungible: {:#?}
+            non_fungible: {:#?}
+            pool_unit: {:#?}
+			"#,
+            self.fungible, self.non_fungible, self.pool_unit,
+        )
+    }
+}
+
+impl HasSampleValues for ResourcePreferences {
+    fn sample() -> Self {
+        Self {
+            fungible: [(ResourceAddress::sample(), EntityFlags::sample())]
+                .into(),
+            non_fungible: [(
+                NonFungibleGlobalId::sample(),
+                EntityFlags::sample(),
+            )]
+            .into(),
+            pool_unit: [(PoolAddress::sample(), EntityFlags::sample())].into(),
+        }
+    }
+
+    fn sample_other() -> Self {
+        Self {
+            fungible: [(
+                ResourceAddress::sample_other(),
+                EntityFlags::sample_other(),
+            )]
+            .into(),
+            non_fungible: [(
+                NonFungibleGlobalId::sample_other(),
+                EntityFlags::sample_other(),
+            )]
+            .into(),
+            pool_unit: [(
+                PoolAddress::sample_other(),
+                EntityFlags::sample_other(),
+            )]
+            .into(),
+        }
+    }
+}
+
+impl ResourcePreferences {
+    pub fn get_hidden_resources(&self) -> HiddenResources {
+        let fungible = self
```

Comment:
You can probably use a closure unlined here, being a function you reuse for the three collections (Rust does not allow nested functions, but you can use closure).